datafusion/sqllogictest/test_files/information_schema.slt

```diff
@@ -603,3 +610,29 @@ query TTTT
 SHOW CREATE TABLE abc;
 ----
 datafusion public abc CREATE EXTERNAL TABLE abc STORED AS CSV LOCATION ../../testing/data/csv/aggregate_test_100.csv
+
+# string_agg has different arg_types but same return type. Test avoiding duplicate entries for the same function.
+query TTT
+select routine_name, data_type, function_type from information_schema.routines where routine_name = 'string_agg';
+----
+string_agg LargeUtf8 AGGREGATE
+
+# test every function type are included in the result
+query TTTTTTTBTTT rowsort
+select * from information_schema.routines where routine_name = 'date_trunc' OR routine_name = 'string_agg' OR routine_name = 'rank';
+----
+datafusion public date_trunc datafusion public date_trunc FUNCTION true Timestamp(Microsecond, None) SCALAR Truncates a timestamp value to a specified precision.
```

Comment:
The type name is an ArrowType name, e.g., `Timestamp (Microsecond, None)`. I prefer to return a SQL type. I think it would be improved if we finished the logical type epic.

Reply:
I agree -- makes sense 👍